packages/@aws-cdk/aws-iotevents/lib/event.ts

```diff
@@ -0,0 +1,16 @@
+/**
+ * Specifies the actions to be performed when the condition evaluates to TRUE.
+ */
+export interface Event {
+  /**
+   * The name of the event
+   */
+  readonly eventName: string;
+
+  /**
+   * The Boolean expression that, when TRUE, causes the actions to be performed.
+   *
+   * @default None - Defaults to perform the actions always.
+   */
+  readonly condition?: string;
```

Comment:
I feel like we need to present an API for these conditions - just writing expressions in strings won't cut it I think.

Reply:
That's sounds great! 😍 
Recently, I knew [`mapping-tenplate.ts` in `aws-appsync`](https://github.com/yamatatsu/aws-cdk/blob/07081ed0b918ecd372b4437ff52a0b0b1ecd021d/packages/@aws-cdk/aws-appsync/lib/mapping-template.ts). I wanna try to design like it! (Or is there any other code I can refer to?)

Reply:
I think that's the closest similar thing we have for this something like this in the CDK currently, yes.

Can you link me to the AWS docs for this functionality? I want to make sure I have a good understanding of the language that's permitted here.

Reply:
Sure! But, as far as I can find, the only document that explains "what a condition is" is [the CloudFormation document](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-iotevents-detectormodel-event.html#:~:text=requires%3A%20No%20interruption-,Condition,not%20a%20Boolean%20value%2C%20the%20actions%20are%20not%20performed%20(%3DFALSE).,-Required%3A%20No).
This CFn document is described that `condition` is expression returning boolean. The refferences of Expressions are [here](https://docs.aws.amazon.com/iotevents/latest/developerguide/iotevents-expressions.html).

And bellow is examples of `condition` from [`Detector model examples `](https://docs.aws.amazon.com/iotevents/latest/developerguide/iotevents-examples.html).

```json
"condition": "!$variable.noDelay && $variable.enteringNewState"
```

```json
"condition": "(((($variable.averageTemperature * ($variable.sensorCount - 1)) + $input.temperatureInput.sensorData.temperature) / $variable.sensorCount) < ($variable.desiredTemperature - $variable.allowedError))"
```

```json
"condition": "currentInput(\"AWS_IoTEvents_Blueprints_Heartbeat_Input\")"
```

```json
"condition": "timeout(\"awake\")"
```